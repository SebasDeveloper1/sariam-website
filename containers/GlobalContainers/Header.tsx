import React from 'react';
import { MdMenu } from 'react-icons/md';
import { Button, IconButton, Typography } from '@/components/index';

export default function Header() {
  return (
    <aside className="">
      <IconButton
        type="button"
        variant="primary"
        styles=""
        title="Menu"
        ariaLabel=""
        icon={MdMenu}
      />
      <Button
        type="button"
        variant="text"
        styles=""
        title=""
        value="aaaaa"
        startIcon={MdMenu}
        endIcon={MdMenu}
      />

      <Typography variant="h1" value="hello world" styles="" />
    </aside>
  );
}
