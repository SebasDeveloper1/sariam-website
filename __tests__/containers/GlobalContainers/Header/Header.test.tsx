// Import the required testing utilities from '@testing-library/react'
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// Import the Header component to be tested
import Header from '@/containers/GlobalContainers/Header';

describe('Header container', () => {
  // Describe the 'Rendering' section of the Header component tests
  describe('Rendering', () => {
    // Test whether the component renders successfully
    test('should render the component successfully', () => {
      const { container } = render(<Header />);
      expect(container).toBeDefined();
    });

    // Test whether the title link is displayed correctly
    test('should show the title link successfully', () => {
      const { getByText } = render(<Header />);
      const link = getByText('SariamParty');
      expect(link).toBeInTheDocument();
    });

    // Test whether the menu button is displayed correctly
    test('should show the menu button successfully', () => {
      render(<Header />);
      const menuButton = screen.getByTitle('Menu');
      expect(menuButton).toBeInTheDocument();
    });

    // Test whether the state of the menu button changes on click
    test('should change the state of the menu button', () => {
      // Arrange
      render(<Header />);
      const menuButton = screen.getByRole('button');

      // Act & Assert
      fireEvent.click(menuButton);
      expect(screen.queryByTitle('Cerrar')).toBeInTheDocument();
      fireEvent.click(menuButton);
      expect(screen.queryByTitle('Menu')).toBeInTheDocument();
    });

    // Test whether the menu opens and closes on clicking the menu button
    test('opens and closes menu when clicking on the menu button', () => {
      render(<Header />);

      // Get the menu button
      const menuButton = screen.getByTitle('Menu');

      // Initially, the menu should be closed
      expect(document.querySelector('.menu-mobile--close')).toBeInTheDocument();
      // Click the menu button to open the menu
      fireEvent.click(menuButton);

      // After clicking, the menu should be open
      expect(document.querySelector('.menu-mobile--open')).toBeInTheDocument();

      // Click the menu button again to close the menu
      fireEvent.click(menuButton);

      // After clicking again, the menu should be closed
      expect(document.querySelector('.menu-mobile--close')).toBeInTheDocument();
    });
    // Test whether the menu items are rendered successfully
    test('should render the menu items successfully', () => {
      render(<Header />);
      const menuItemsXl = document.querySelectorAll('.navbar-item-xl');
      expect(menuItemsXl).toBeTruthy();
      const menuItemsSm = document.querySelectorAll('.navbar-item-sm');
      expect(menuItemsSm).toBeTruthy();
    });
  });
  describe('Snapshots', () => {
    // Test if the rendered output matches the previous snapshot
    test('should match snapshot', () => {
      const { container } = render(<Header />);
      expect(container).toMatchSnapshot();
    });
  });
});
