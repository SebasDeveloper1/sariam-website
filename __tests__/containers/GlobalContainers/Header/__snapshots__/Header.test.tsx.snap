// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header container Snapshots should match snapshot 1`] = `
<div>
  <header
    class="z-20 fixed top-0 flex items-center w-full h-16 bg-gray-900 shadow-sm shadow-gray-950"
  >
    <nav
      class="flex justify-between w-11/12 mx-auto"
    >
      <div
        class="flex items-center justify-between w-full md:gap-x-12"
      >
        <a
          class="className font-bold text-3xl text-white"
          href="#inicio"
        >
          SariamParty
        </a>
        <ul
          class="hidden xl:flex items-center justify-center gap-x-6"
        >
          <li
            class="navbar-item-xl"
          >
            <a
              href="#inicio"
            >
              Inicio
            </a>
          </li>
          <li
            class="navbar-item-xl"
          >
            <a
              href="#eventos"
            >
              Últimos eventos
            </a>
          </li>
          <li
            class="navbar-item-xl"
          >
            <a
              href="#servicios"
            >
              Servicios
            </a>
          </li>
          <li
            class="navbar-item-xl"
          >
            <a
              href="#opiniones"
            >
              Opiniones
            </a>
          </li>
          <li
            class="button-primary rounded-2xl"
          >
            <a
              href="#contacto"
            >
              Contacto
            </a>
          </li>
        </ul>
        <button
          aria-label="Menu"
          class="button-text rounded-full p-2.5 xl:hidden transition-all duration-300 transform text-lime-500 hover:text-lime-400  "
          title="Menu"
          type="button"
        >
          <span
            class="text-4xl"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <div
          class="fixed top-0 xl:hidden inset-0 h-screen flex justify-center px-10 py-20 mt-16 bg-gray-900/70 backdrop-filter backdrop-blur-md transition-all duration-500 ease-in-out menu-mobile--close"
        >
          <ul
            class="flex flex-col items-center gap-y-6 w-full"
          >
            <li
              class="navbar-item-sm w-full"
            >
              <a
                href="#inicio"
              >
                Inicio
              </a>
            </li>
            <li
              class="navbar-item-sm w-full"
            >
              <a
                href="#eventos"
              >
                Últimos eventos
              </a>
            </li>
            <li
              class="navbar-item-sm w-full"
            >
              <a
                href="#servicios"
              >
                Servicios
              </a>
            </li>
            <li
              class="navbar-item-sm w-full"
            >
              <a
                href="#opiniones"
              >
                Opiniones
              </a>
            </li>
            <li
              class="navbar-item-sm w-full"
            >
              <a
                href="#contacto"
              >
                Contacto
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </header>
</div>
`;
